app/services/site_configs/upsert.rb

```diff
@@ -0,0 +1,135 @@
+module SiteConfigs
+  class Upsert
+    EMOJI_ONLY_FIELDS = %w[community_emoji].freeze
+    IMAGE_FIELDS =
+      %w[
+        main_social_image
+        logo_png
+        secondary_logo_url
+        campaign_sidebar_image
+        mascot_image_url
+        mascot_footer_image_url
+        onboarding_logo_image
+        onboarding_background_image
+        onboarding_taskcard_image
+      ].freeze
+
+    VALID_URL = %r{\A(http|https)://([/|.|\w|\s|-])*.[a-z]{2,5}(:[0-9]{1,5})?(/.*)?\z}.freeze
+
+    VALID_DOMAIN = /^[a-zA-Z0-9][a-zA-Z0-9-]{1,61}[a-zA-Z0-9](?:\.[a-zA-Z]{2,})+$/.freeze
+
+    attr_reader :errors
+
+    def self.call(configs)
+      new(configs).call
+    end
+
+    def initialize(configs)
+      @configs = configs
+      @success = false
+    end
+
+    def call
+      @errors = []
+      clean_up_params
+      validate_inputs
+      validate_emoji
+      validate_image_urls
+
+      return self if @errors.flatten.any?
+
+      @success = true
+      @configs.each do |key, value|
```

Comment:
**suggestion (non-blocking):** I'd probably move this out into a private helper like `upsert_configs`, similar to the methods at the beginning of this one.

Reply:
👍